【cherry-pick】Optimize imports and fix NameError in static/ppdet/modeling/tests/test_architectures.py

Mandroide opened the pull request:
#3557

yghstill commented on ppdet/utils/checkpoint.py:
```diff
@@ -20,8 +20,7 @@
 import os
 import time
-import re
-import numpy as np
```
Don't delete numpy

yghstill commented on static/deploy/python/infer.py:
```diff
@@ -28,12 +30,11 @@
 import numpy as np
 import paddle
 import paddle.fluid as fluid
-from preprocess import preprocess, Resize, Normalize, Permute, PadStride
```
Don't delete Resize etc

Mandroide committed:
Optimize imports

Besides removing unnecessary imports, this also fixes the missing logger module in test_architectures.py within the static directory.
